### ISSUE TITLE
Relative volume

### ISSUE THREAD

**Issue by torerlings** [roelofoomen/com.bluesound#4]
I’m having trouble adjusting relative volume on my powernode 2i. No matter what percentage up or down I choose, the result is that the powernode mutes the volume.

**Comment by jghaanstra** [comment]
I have made some changes in the master branch here on GitHub that hopefully fixes this. Do you know how to run the app from the command line as described here: https://community.athom.com/t/how-to-cli-install-method/198

Master: https://github.com/jghaanstra/com.bluesound

**Comment by torerlings** [comment]
Hi again, and thanks for the quick response. I tried the updated version, and now it doesn't mute anymore, but it only adjust the volume upwards. As far as I can see the percentage adjusted is the same, but again, it only gets louder

**Comment by jghaanstra** [comment]
Could you explain a bit more which cards with which configuration you are using to mute and change relative volume. I don't own a Bluesound device myself so I can't test anything. I need more specific details on what you did, what the expected result should be and what the actual result was.

**Comment by torerlings** [comment]
Hi again. Right now, I have just tried the relative volume card for controlling volume from a Heatit Z-push button 8 panel. I didn't have much time to test yesterday, but in the flow I was testing (when button x is pushed, then relative volume -50%) it seemed like the first time I pushed it the volume very briefly went down, and then up again, the next time i pushed it it adjusted the volume 50% upwards. It seems like the same flow only with +50% instead of -50% works fine. Note, this is only observed in the bluesound app, it was in the middle of the night so I couldn't risk waking up the whole house. :) Hope this answers your question?

**Comment by jghaanstra** [comment]
Ok, I pushed some new changes to the master branch here on GitHub. Could you redownload and try again and let me know the result.

**Comment by torerlings** [comment]
Hi again. Now the relative volume card doesn't seem to do anything. It confirms when I test the flow in the homey app, but nothing happens. When I test the mute and unmute card it just keeps spinning and never confirms, nothing happens on the bluesound product.

**Comment by jghaanstra** [comment]
Uggh, what a hassle to get this right without the actual device. Just by looking at the code I can not explain this. I have added extra debug logging in the master branch to help me figure it out. Could you use `homey app run`, try and change the relative volume and post the output from the console here.

**Comment by torerlings** [comment]
Sorry about the late reply. Here I attached the output where I tried 3 times to set relative volume .20%, but nothing happens in Bluesound. Does this tell you anything, and is there anything else you want me to try?

![homey bluesound](https://user-images.githubusercontent.com/57958571/69923544-050baf80-14a6-11ea-9d8d-bea01384322c.PNG)

**Comment by jghaanstra** [comment]
Ok thanx. I just updated the master branch. Could you check again if it's fixed now.

**Comment by torerlings** [comment]
Hi again, and again sorry for late reply. I have only tested a little bit, but it seems to work now. The only thing, and this could perhaps be a Homey issue and not an app issue, is that I have set it to adjust the volume in relatively small increments, like a regular remote control basically, but I experience that I have to wait a bit before i can trigger the flow a second or third time. Is this something you think could be fixed, or is this an inherent delay in the homey solution?
